store/sql_store.go

```diff
@@ -56,9 +57,30 @@ func NewSqlStore() Store {
 			utils.Cfg.SqlSettings.Trace)
 	}
 
+	schemaVersion := sqlStore.GetCurrentSchemaVersion()
+
+	// If the version is already set then we are potentially in an 'upgrade needed' state
+	if schemaVersion != "" {
+		// Check to see if it's the most current database schema version
+		if !model.IsCurrentVersion(schemaVersion) {
+			// If we are upgrading from the previous version then print a warning and continue
+			if model.IsPreviousVersion(schemaVersion) {
+				l4g.Warn("The database schema version of " + schemaVersion + " appears to be out of date")
+				l4g.Warn("Attempting to upgrade the database schema version to " + model.CurrentVersion)
+			} else {
+				// If this is an 'upgrade needed' state but the user is attempting to skip a version then halt the world
+				l4g.Critical("The database schema version of " + schemaVersion + " cannot be upgraded.  You must not skip a version.")
+				time.Sleep(time.Second)
+				panic("The database schema version of " + schemaVersion + " cannot be upgraded.  You must not skip a version.")
+			}
+		}
+	}
+
 	// Temporary upgrade code, remove after 0.8.0 release
-	if sqlStore.DoesColumnExist("Sessions", "AltId") {
-		sqlStore.GetMaster().Exec("DROP TABLE IF EXISTS Sessions")
+	if sqlStore.DoesTableExist("Sessions") {
+		if sqlStore.DoesColumnExist("Sessions", "AltId") {
+			sqlStore.GetMaster().Exec("DROP TABLE IF EXISTS Sessions")
```

Comment:
This query can lose the "IF EXISTS" part but doesn't matter since it's temp code anyway